A New Hope/Day1Main.py

```diff
@@ -0,0 +1,12 @@
+from Day1.BuildingNavigator import BuildingNavigator
+
+with open("input1.txt", "r") as input_file:
+    instructions = input_file.read()
+
+building_navigator = BuildingNavigator(instructions)
+building_navigator.traverse_building_to_end_of_input()
+print(f"Santa arrives at floor {building_navigator.santas_current_floor}.")
```

Comment:
Ezt temporal coupling-nak hivjak, amikor a class-od egyes funkcioi csak akkor mukodnek helyesen ha adott sorrendben hivod meg a dolgokat. 
Meaning, ha kicsit atrendeznem ezt a file-t es mondjuk a vegen akarnam kiirni mind a ket sor megoldast (for whatever requirement) akkor mar nem lesz helyes. 
Erre az lenne a jo megoldas hogy nem a class attribute-umjat olvasod, hanem a fuggvenyek visszaadjak a megoldast. 

Ez amiatt is fontos, mert nem jo practice data-t expose-olni. A data az implementacios reszlet, arrol nem szabad senkinek sem tudja hogy te hogyan reprezentaltad az adatokat class-on belul. 
Ha holnap valtozik a requirement es valamiert mas adat reprezentaciot kell valasztanod, akkor at kell irnod az osszes kodot ami ezt (pl santas current floor) hasznalta. 

Sot, ha engem kerdezel akkor az a legfontosabb dolga egy class-nak hogy elrejtsen minden adatot es csak funkcionalitast (methods) expose-oljon. 

Ha ragaszkodsz hozza hogy "current floor"-t kerjen el a class-tol, akkor ide pont kell a "get_current_floor" method, de en inkabb azt mondanam hogy adja vissza az erteket a traverse_building_to_end_of_input

Reply:
Ja meg annyi jutott eszembe, lehet ez nem tiszta, de amikor egy class "josagarol" beszelunk, akkor az nem csak arrol szol hogy mi a neve a attributumoknak meg mik a fuggvenyek. Az is resze (szar fuggvenyekkel nehez jo class-t csinalni), de elso sorban az a lenyeg hogy jol valasztottad-e ki a reprezentaciot. 

Ha egyszerre mindent csinal, akkor nem lehet jo a class, mindegy hogy mik a fuggvenyei meg az attribute-ok. Ha nagyon sok requirement valtozas miatt kell atirnod a class-t akkor nem lehet jo. Ha olyan fuggvenyekbol all amiknek nincs egymashoz kozuk, akkor nem lehet jo. Ha tul tightly coupled mas osztalyokhoz (ez nalad meg nem jelent meg mint problema mert nincsenek is mas osztalyaid :D ) akkor nem lehet jo class. 

A formatum meg oroklodes meg ilyenek (amit gondolom az EPAM-os kurzus elmondott) azok csak a beugrok, de koncepcionalisan jol kell illeszkedjenek a class-aid a problema korhoz. Ahogy egyszer mar emlitettem, leginkabb az adatok hatarozzak meg hogy mi lesz egy class. Egy vagy tobb data point reprezentacioja egy class, elrejti azokat az adatokat es olyan operation-oket biztosit (a fuggvenyek reven) amikhez amugy szukseged lenne az adatra.